src/app/vulnerabilities/Drupal Username Enumeration.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+use Psr\Http\Message\ResponseInterface as Response;
+use Psr\Http\Message\ServerRequestInterface as Request;
+use Slim\Factory\AppFactory;
+
+$app->get(
+    '/user/1',
+    function (Request $request, Response $response) {
+        $html = file_get_contents(__DIR__ . "/../resources/user/admin.html");
+        $response->getBody()->write($html);
+        return $response->withHeader("location", "/user/admin")
+                        ->withStatus(301);
+    }
+);
+
+$app->get(
+    '/user/2',
+    function (Request $request, Response $response) {
+        $html = file_get_contents(__DIR__ . "/../resources/user/karthik.html");
+        $response->getBody()->write($html);
+        return $response->withHeader("location", "/user/karthik")
+                        ->withStatus(301);
+    }
+);
+
+$app->get(
+    '/user/3',
+    function (Request $request, Response $response) {
+        $html = file_get_contents(__DIR__ . "/../resources/user/prince.html");
+        $response->getBody()->write($html);
+        return $response->withHeader("location", "/user/prince")
+                        ->withStatus(301);
+    }
+);
+
+$app->get(
+    '/guest/password',
+    function (Request $request, Response $response) {
+        $html = file_get_contents(__DIR__ . "/../resources/user/forbidden.html");
```

Comment:
Have a look at src/app/index.php for an idea on how to render simple html directly in the php file, without importing a dedicated html file.

Reply:
Yes, I've implemented that now.